packages/transport-webrtc/src/webrtc/rtc-peer-connection.ts

```diff
@@ -0,0 +1,306 @@
+import node from 'node-datachannel'
+import defer, { type DeferredPromise } from 'p-defer'
+import { DataChannel } from './rtc-data-channel.js'
+import { DataChannelEvent, PeerConnectionIceEvent } from './rtc-events.js'
+import { IceCandidate } from './rtc-ice-candidate.js'
+import { SessionDescription } from './rtc-session-description.js'
+
+export class PeerConnection extends EventTarget implements RTCPeerConnection {
+  static async generateCertificate (keygenAlgorithm: AlgorithmIdentifier): Promise<RTCCertificate> {
+    throw new Error('Not implemented')
+  }
+
+  canTrickleIceCandidates: boolean | null
+  sctp: RTCSctpTransport | null
+
+  onconnectionstatechange: ((this: RTCPeerConnection, ev: Event) => any) | null
+  ondatachannel: ((this: RTCPeerConnection, ev: RTCDataChannelEvent) => any) | null
+  onicecandidate: ((this: RTCPeerConnection, ev: RTCPeerConnectionIceEvent) => any) | null
+  onicecandidateerror: ((this: RTCPeerConnection, ev: Event) => any) | null
+  oniceconnectionstatechange: ((this: RTCPeerConnection, ev: Event) => any) | null
+  onicegatheringstatechange: ((this: RTCPeerConnection, ev: Event) => any) | null
+  onnegotiationneeded: ((this: RTCPeerConnection, ev: Event) => any) | null
+  onsignalingstatechange: ((this: RTCPeerConnection, ev: Event) => any) | null
+  ontrack: ((this: RTCPeerConnection, ev: RTCTrackEvent) => any) | null
+
+  #peerConnection: node.PeerConnection
+  #config: RTCConfiguration
+  #localOffer: DeferredPromise<RTCSessionDescriptionInit>
+  #localAnswer: DeferredPromise<RTCSessionDescriptionInit>
+  #dataChannels: Set<DataChannel>
+
+  constructor (init: RTCConfiguration = {}) {
+    super()
+
+    this.#config = init
+    this.#localOffer = defer()
+    this.#localAnswer = defer()
+    this.#dataChannels = new Set()
+
+    const iceServers = init.iceServers ?? []
+
+    this.#peerConnection = new node.PeerConnection(`peer-${Math.random()}`, {
+      iceServers: iceServers.map(server => {
+        const urls = (Array.isArray(server.urls) ? server.urls : [server.urls]).map(str => new URL(str))
+
+        return urls.map(url => {
+          /** @type {import('../lib/index.js').IceServer} */
+          const iceServer = {
+            hostname: url.hostname,
+            port: parseInt(url.port, 10),
+            username: server.username,
+            password: server.credential
+            // relayType - how to specify?
```

Comment:
Looks like the [Relay type](https://github.com/murat-dogan/node-datachannel/blob/9e35a2040a569e61ba98231cd37e9232b95bf656/API.md?plain=1#L24-L28) is an enum specified by a string?

Reply:
Yes, but the `RTCIceServer` type from the standard doesn't have a `relayType` field so there's nothing to map this from.